components/identities/actions/autoRenewalModal.tsx

```diff
@@ -0,0 +1,308 @@
+    // register the metadata to the sales manager db
+    // when enabling auto renewal, if user wasn't already registered
+    if (needMedadata) {
+      fetch(`${process.env.NEXT_PUBLIC_SALES_SERVER_LINK}/add_metadata`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          meta_hash: metadataHash,
+          email,
+          tax_state: isUsResident ? usState : "none",
+          salt: salt,
+        }),
+      })
+        .then((res) => res.json())
+        .catch((err) => console.log("Error while sending metadata:", err));
+    }
+
+    fetch(`${process.env.NEXT_PUBLIC_SALES_SERVER_LINK}/mail_subscribe`, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        tx_hash: formatHexString(autorenewData.transaction_hash),
+        groups,
+      }),
+    })
+      .then((res) => res.json())
+      .catch((err) => console.log("Error on registering to email:", err));
+
+    addTransaction({
+      timestamp: Date.now(),
+      subtext: `Enabled auto renewal for ${domain}`,
+      type: NotificationType.TRANSACTION,
+      data: {
+        type: TransactionType.ENABLE_AUTORENEW,
+        hash: autorenewData.transaction_hash,
+        status: "pending",
+      },
+    });
+    setIsTxSent(true);
+  }, [autorenewData, usState, salt]);
+
+  function changeEmail(value: string): void {
+    setEmail(value);
+    setEmailError(isValidEmail(value) ? false : true);
+  }
+
+  return (
+    <Modal
+      disableAutoFocus
+      open={isModalOpen}
+      onClose={handleClose}
+      aria-labelledby="modal-modal-title"
+      aria-describedby="modal-modal-description"
+    >
+      {isTxSent ? (
+        <ConfirmationTx
+          closeModal={handleClose}
+          txHash={autorenewData?.transaction_hash}
+        />
+      ) : (
+        <div className={`${styles.menu} "overflow-scroll"`}>
+          <button className={styles.menu_close} onClick={handleClose}>
+            <svg viewBox="0 0 24 24">
+              <path
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                strokeWidth="2"
+                d="M6 18L18 6M6 6l12 12"
+              ></path>
+            </svg>
+          </button>
+          <div className={styles.form}>
+            <div className="flex flex-col items-start gap-4 self-center">
+              <h2 className={styles.title}>Enable auto renewal</h2>
+            </div>
+            <div className="flex flex-col items-start gap-4 self-center">
+              <p className={styles.desc}>
+                Enable automatic domain renewal to ensure uninterrupted
+                ownership and benefits. Never worry about expiration dates
+                again.{" "}
+                {identity?.domain_expiry
+                  ? `Your domain will be renewed for 1 year on  
+                ${timestampToReadableDate(
+                  identity.domain_expiry - 2592000
```

Comment:
I guess `2592000` is the second equivalent to a month. We need to put that in a const for us to understand easily